core/jvm/src/main/scala/zio/internal/ZScheduler.scala

```diff
@@ -200,74 +198,58 @@ private final class ZScheduler(autoBlocking: Boolean) extends Executor {
           }
           notify = true
         } else {
-          globalQueue.offerAll(worker.localQueue.pollUpTo(128) :+ runnable)
+          val rnd = ThreadLocalRandom.current()
+          globalQueue.offerAll(worker.localQueue.pollUpTo(128), rnd)
+          globalQueue.offer(runnable, rnd)
```

Comment:
🟢 Minor

I see here that you changed one call to `.offerAll` with two calls: 1 call to `.offerAll` and 1 call to `.offer`
Each call to `.offerAll` and `.offer` will call `random.nextInt(nQueues)`
Which means that replacing this 1 `.offerAll` call with 1 `.offerAll` and 1 `.offer` will generate 2 calls to `random.nextInt(nQueues)`. 
AFAIK, appending to a Chunk is an optimized operation (ie. https://zio.dev/reference/stream/chunk/#high-performance).

How expensive is a call to `random.nextInt(nQueues)`? If more expensive than appending, would it be worth it keep the unique `globalQueue.offerAll(worker.localQueue.pollUpTo(128) :+ runnable, rnd)` call? 🤔

Reply:
Now that I think a bit more about it, I believe we should be offering the `runnable` to the local queue instead. This way we're to some degree preserving the ordering that the runnables are meant to be enqueued. 

What do you think about that?

Reply:
I think it makes more sense to offer to the global queue. It's supposed to be executed after those 128 we just enqueued so it makes sense to be offered after. Also if you offer to local, it could potentially fail again so it complicated the flow.

Reply:
> It's supposed to be executed after those 128 we just polled/enqueued

This is why I think it's meant to be placed in the local queue; it's _not_ meant to be executed after those 128 we just polled. It's meant to be executed after the remaining 128 in the local queue that were previously at the tail of it and now at start. If we place it in the global queue, we're breaking the sequence

Ideally, we'd want to dequeue the last 128 from the local queue, but that would be bad for performance

Reply:
Ah you're right, the 128 we dequeue should be executed first, not last. I agree with you then, but you still need to handle the worst case scenario where it's full again.

Reply:
So while a worker can steal from another worker's local queue, only the worker that owns the queue can add elements to its local one. So unless some bug comes in it _shouldn't_ be possible for this to happen.

Having said that, bugs always find a way in, so I'll use this _just in case_:

```scala
if (!worker.localQueue.offer(runnable)) globalQueue.offer(runnable)
```